Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -133,9 +133,11 @@ class TopSitesPanel: UIViewController {
         collection?.indexPathsForVisibleItems().forEach(updateRemoveButtonStateForIndexPath)
     }
 
-    private func deleteTileForSuggestedSite(site: SuggestedSite) -> Success {
+    private func deleteTileForSuggestedSite(site: Site) -> Success {
         var deletedSuggestedSites = profile.prefs.arrayForKey("topSites.deletedSuggestedSites") as! [String]
-        deletedSuggestedSites.append(site.url)
+        if !deletedSuggestedSites.contains(NSURL(string: site.url)?.normalizedHost() ?? site.url) {
```

Comment:
You have `NSURL(string: site.url)?.normalizedHost() ?? site.url` repeated in lots of places here. This should be factored out into a helper method.